src/ECharts/Utils.purs

```diff
@@ -11,4 +20,322 @@ i.e
+  converter_MMMdDD dateTime = 
+    case DT.fromString dateTime of
+      Just dt -> (take 3 (show $ month dt)) ++ "-" ++
+        (replace ")" "" (replace "(DayOfMonth " "" (show $ dayOfMonth dt)))
+      _ -> dateTime
+
+  converter_MMdDD :: String -> String
+  converter_MMdDD dateTime = 
+    case DT.fromString dateTime of
+      Just dt ->
+        case month dt of
+          DT.January -> "01"
+          DT.February -> "02"
+          DT.March -> "03"
+          DT.April -> "04"
+          DT.May -> "05"
+          DT.June -> "06"
+          DT.July -> "07"
+          DT.August -> "08"
+          DT.September -> "09"
+          DT.October -> "10"
+          DT.November -> "11"
+          DT.December -> "12"
+        ++ "-" ++ (replace ")" "" (replace "(DayOfMonth " "" (show $ dayOfMonth dt)))
+      _ -> dateTime
+
+  converter_YYYYdMMMdDD :: String -> String
+  converter_YYYYdMMMdDD dateTime = 
+    case DT.fromString dateTime of
+      Just dt -> (replace ")" "" (replace "(Year " "" (show $ year dt))) ++ "-" ++
+        (take 3 (show $ month dt)) ++ "-" ++
+        (replace ")" "" (replace "(DayOfMonth " "" (show $ dayOfMonth dt)))
+      _ -> dateTime
+
+  converter_YYYYdMMdDD :: String -> String
+  converter_YYYYdMMdDD dateTime = 
+    case DT.fromString dateTime of
+      Just dt -> replace ")" "" (replace "(Year " "" (show $ year dt)) ++ "-" ++
+        case month dt of
+          DT.January -> "01"
+          DT.February -> "02"
+          DT.March -> "03"
+          DT.April -> "04"
+          DT.May -> "05"
+          DT.June -> "06"
+          DT.July -> "07"
+          DT.August -> "08"
+          DT.September -> "09"
+          DT.October -> "10"
+          DT.November -> "11"
+          DT.December -> "12"
+        ++ "-" ++ (replace ")" "" (replace "(DayOfMonth " "" (show $ dayOfMonth dt)))
+      _ -> dateTime
+
+  converter_MMMsDDsYYYY :: String -> String
+  converter_MMMsDDsYYYY dateTime = 
+    case DT.fromString dateTime of
+      Just dt -> (take 3 (show $ month dt)) ++ "/" ++
+      	(replace ")" "" (replace "(DayOfMonth " "" (show $ dayOfMonth dt))) ++ "/" ++
+      	(replace ")" "" (replace "(Year " "" (show $ year dt))) 
+      _ -> dateTime
+
+  doNothing :: String -> String
+  doNothing dateTime = dateTime
+
+
+{-
+This func is used to format numeral string accoridng to some template.
+Supported Templates:
+"0": 1
+"0.0": 1.1
+"0.00": 1.12
+"0.000": 1.123
+"0.0000": 1.1234
+"0e": 1e+1
+"0.0e": 1.1e+1
+"0.00e" 1.12e+1
+"0a": 1K
+"0.0a": 1.1K
+"0.00a": 1.12K
+"0%": 1%
+"0.0%": 1.1%
+"0.00%": 1.12%
+-}
+numeralFormatter :: String -> (Number -> String)
+numeralFormatter template = numeralFormatterWithValMnplt 0.0 1.0 template
+
+
+{-
+This func can
+1) manipluate numerical values by adding and then multiplying
+2) format the manipulated value to a string according to some template
+-}
+numeralFormatterWithValMnplt :: Number -> Number -> String -> (Number -> String)
+numeralFormatterWithValMnplt add mul template = 
```

Comment:
@cryogenian The param of function `numeralFormatter :: String -> (Number -> String)` is a string, which represents the formate template, rather than a number value, I don't know how to manipulate the number param of the returned callback function by applying function composition. The way listed above doesn't seems to work.